services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ConnectedDevicesSpec.groovy

```diff
@@ -637,21 +669,144 @@ srcLldpDevices=#newSrcEnabled, dstLldpDevices=#newDstEnabled"() {
             }
         }
 
-        then: "Device is also visible as a flow device"
+        and: "Device is also visible as a flow device"
+        with(northbound.getFlowConnectedDevices(flow.id)) {
+            it.source.lldp.size() == 1
+            verifyEquals(it.source.lldp.first(), lldpData)
+        }
+
+        and: "Device is NOT visible as a default flow device"
+        northbound.getFlowConnectedDevices(defaultFlow.id).source.lldp.empty
+
+        when: "Another device sends an lldp packet into a flow port with vlan different from flow vlan"
+        def lldpData2 = LldpData.buildRandom()
+        def vlan = flow.source.vlanId - 1
+        new ConnectedDevice(traffExamProvider.get(), tg, vlan).withCloseable {
+            it.sendLldp(lldpData2)
+        }
+
+        then: "Corresponding device is detected on a switch port with reference to corresponding default flow"
         Wrappers.wait(WAIT_OFFSET) {
-            with(northbound.getFlowConnectedDevices(flow.id)) {
-                it.source.lldp.size() == 1
-                verifyEquals(it.source.lldp.first(), lldpData)
+            with(northboundV2.getConnectedDevices(sw.dpId).ports) {
+                it.size() == 1
+                it[0].portNumber == tg.switchPort
+                it[0].lldp.size() == 2
+                it[0].lldp.last().flowId == defaultFlow.id
+                //this behavior explained in 'where' block comment
```

Comment:
I don't see any comment in the `where`  block

Reply:
removed the comment. It used to describe some weird behavior here which no longer exists